// @elasticemail/elasticemail-client/dist/model/SmtpCredentials.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AccessLevel = _interopRequireDefault(require("./AccessLevel"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
/**
 * The SmtpCredentials model module.
 * @module model/SmtpCredentials
 * @version 4.0.23
 */
var SmtpCredentials = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>SmtpCredentials</code>.
   * SMTP Credentials info
   * @alias module:model/SmtpCredentials
   */
  function SmtpCredentials() {
    _classCallCheck(this, SmtpCredentials);
    SmtpCredentials.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(SmtpCredentials, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>SmtpCredentials</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/SmtpCredentials} obj Optional instance to populate.
     * @return {module:model/SmtpCredentials} The populated <code>SmtpCredentials</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new SmtpCredentials();
        if (data.hasOwnProperty('AccessLevel')) {
          obj['AccessLevel'] = _AccessLevel["default"].constructFromObject(data['AccessLevel']);
        }
        if (data.hasOwnProperty('Name')) {
          obj['Name'] = _ApiClient["default"].convertToType(data['Name'], 'String');
        }
        if (data.hasOwnProperty('DateCreated')) {
          obj['DateCreated'] = _ApiClient["default"].convertToType(data['DateCreated'], 'Date');
        }
        if (data.hasOwnProperty('LastUse')) {
          obj['LastUse'] = _ApiClient["default"].convertToType(data['LastUse'], 'Date');
        }
        if (data.hasOwnProperty('Expires')) {
          obj['Expires'] = _ApiClient["default"].convertToType(data['Expires'], 'Date');
        }
        if (data.hasOwnProperty('RestrictAccessToIPRange')) {
          obj['RestrictAccessToIPRange'] = _ApiClient["default"].convertToType(data['RestrictAccessToIPRange'], ['String']);
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>SmtpCredentials</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>SmtpCredentials</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Name'] && !(typeof data['Name'] === 'string' || data['Name'] instanceof String)) {
        throw new Error("Expected the field `Name` to be a primitive type in the JSON string but got " + data['Name']);
      }
      // ensure the json data is an array
      if (!Array.isArray(data['RestrictAccessToIPRange'])) {
        throw new Error("Expected the field `RestrictAccessToIPRange` to be an array in the JSON data but got " + data['RestrictAccessToIPRange']);
      }
      return true;
    }
  }]);
  return SmtpCredentials;
}();
/**
 * @member {module:model/AccessLevel} AccessLevel
 */
SmtpCredentials.prototype['AccessLevel'] = undefined;

/**
 * Name of the key.
 * @member {String} Name
 */
SmtpCredentials.prototype['Name'] = undefined;

/**
 * Date this SmtpCredential was created.
 * @member {Date} DateCreated
 */
SmtpCredentials.prototype['DateCreated'] = undefined;

/**
 * Date this SmtpCredential was last used.
 * @member {Date} LastUse
 */
SmtpCredentials.prototype['LastUse'] = undefined;

/**
 * Date this SmtpCredential expires.
 * @member {Date} Expires
 */
SmtpCredentials.prototype['Expires'] = undefined;

/**
 * Which IPs can use this SmtpCredential
 * @member {Array.<String>} RestrictAccessToIPRange
 */
SmtpCredentials.prototype['RestrictAccessToIPRange'] = undefined;
var _default = SmtpCredentials;
exports["default"] = _default;
