// @elasticemail/elasticemail-client/dist/model/Utm.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
/**
 * The Utm model module.
 * @module model/Utm
 * @version 4.0.23
 */
var Utm = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>Utm</code>.
   * Utm marketing data to be attached to every link in this e-mail.
   * @alias module:model/Utm
   */
  function Utm() {
    _classCallCheck(this, Utm);
    Utm.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(Utm, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>Utm</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/Utm} obj Optional instance to populate.
     * @return {module:model/Utm} The populated <code>Utm</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new Utm();
        if (data.hasOwnProperty('Source')) {
          obj['Source'] = _ApiClient["default"].convertToType(data['Source'], 'String');
        }
        if (data.hasOwnProperty('Medium')) {
          obj['Medium'] = _ApiClient["default"].convertToType(data['Medium'], 'String');
        }
        if (data.hasOwnProperty('Campaign')) {
          obj['Campaign'] = _ApiClient["default"].convertToType(data['Campaign'], 'String');
        }
        if (data.hasOwnProperty('Content')) {
          obj['Content'] = _ApiClient["default"].convertToType(data['Content'], 'String');
        }
      }
      return obj;
    }

    /**
     * Validates the JSON data with respect to <code>Utm</code>.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @return {boolean} to indicate whether the JSON data is valid with respect to <code>Utm</code>.
     */
  }, {
    key: "validateJSON",
    value: function validateJSON(data) {
      // ensure the json data is a string
      if (data['Source'] && !(typeof data['Source'] === 'string' || data['Source'] instanceof String)) {
        throw new Error("Expected the field `Source` to be a primitive type in the JSON string but got " + data['Source']);
      }
      // ensure the json data is a string
      if (data['Medium'] && !(typeof data['Medium'] === 'string' || data['Medium'] instanceof String)) {
        throw new Error("Expected the field `Medium` to be a primitive type in the JSON string but got " + data['Medium']);
      }
      // ensure the json data is a string
      if (data['Campaign'] && !(typeof data['Campaign'] === 'string' || data['Campaign'] instanceof String)) {
        throw new Error("Expected the field `Campaign` to be a primitive type in the JSON string but got " + data['Campaign']);
      }
      // ensure the json data is a string
      if (data['Content'] && !(typeof data['Content'] === 'string' || data['Content'] instanceof String)) {
        throw new Error("Expected the field `Content` to be a primitive type in the JSON string but got " + data['Content']);
      }
      return true;
    }
  }]);
  return Utm;
}();
/**
 * utmsource value
 * @member {String} Source
 */
Utm.prototype['Source'] = undefined;

/**
 * utmmedium value
 * @member {String} Medium
 */
Utm.prototype['Medium'] = undefined;

/**
 * utmcampaign value
 * @member {String} Campaign
 */
Utm.prototype['Campaign'] = undefined;

/**
 * utmcontent value
 * @member {String} Content
 */
Utm.prototype['Content'] = undefined;
var _default = Utm;
exports["default"] = _default;
